app/app_software.go

```diff
@@ -13,3 +16,9 @@ import (
 func NewWithID(id string) fyne.App {
 	return newAppWithDriver(test.NewDriverWithPainter(software.NewPainter()), id)
 }
+
+// NewWithSoftwareDriver returns a new app instance using the Software (custom) driver.
+// The ID string should be globally unique to this app.
+func NewWithSoftwareDriver(id string, painter func(image.Image), events chan any) fyne.App {
```

Comment:
Won't the painter always be the one we have in the software package?

Reply:
TBH I didn't really look at the logic for how that worked, but I needed a way to add some extra params and wasn't sure a good way to do that with the existing interfaces.